### PR TITLE
Modernize `NSDate` and `Date` implementation introducing breaking changes

### PR DESCRIPTION
**Pull request by mokagio**
Follows up on @crazytonyli comments from the PR https://github.com/wordpress-mobile/WordPressKit-iOS/pull/758, which it targets.

I don't know why I felt like making a dedicated PR instead of pushing to that one directly 🤷‍♂️ Maybe because of the breaking changes...

---

- [ ] Please check here if your pull request includes additional test coverage. — N.A.
- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary. — N.A.

### CODE REVIEW

File: WordPressKit/Date+WordPressCom.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    /// - SeeAlso: [WordPress.com REST API docs](https://developer.wordpress.com/docs/api/)
+    /// - Warning: This method doesn't support fractional seconds or dates with leap seconds (23:59:60 turns into 23:59:00)
+    static func with(wordPressComJSONString jsonString: String) -> Date? {
+        DateFormatter.wordPressCom.date(from: jsonString)
+    }
+
+    var wordPressComJSONString: String {
+        DateFormatter.wordPressCom.string(from: self)
+    }
+}
```
Comment (crazytonyli) [inline WordPressKit/Date+WordPressCom.swift:19]: I feel like these extension functions don't really help much. They can just use the formatter instance `DateFormatter.wordPressCom` directly, right?

Comment (crazytonyli) [reply WordPressKit/Date+WordPressCom.swift:19]: Sorry, I just noticed `wordPressCom` is an internal API. Please ignore my comment above, if you intend to not expose the `DateFormatter` as public API, which I think is a good idea (leaving us option to switch over to [ISO8601DateFormatter](https://developer.apple.com/documentation/foundation/iso8601dateformatter/options/1643217-withinternetdatetime)).

**Review by crazytonyli — APPROVED** [review]
I'm okay with merging these two PR together as one. It might be easier for code review 🙈

File: WordPressKit/Date+WordPressCom.swift (hunk 1 above)
Comment (mokagio) [reply WordPressKit/Date+WordPressCom.swift:19]: 👍 

Yes. This whole custom formatting feels odd. I think once we get the code in a better place in term of SPM, we could look at trimming some of those unnecessary custom bits in favor of standards

**Pull request merged**